src/objectExplorer/objectExplorerService.ts

```diff
@@ -350,18 +353,40 @@ export class ObjectExplorerService {
 
     public async removeObjectExplorerNode(node: TreeNodeInfo, isDisconnect: boolean = false): Promise<void> {
         await this.closeSession(node);
+        const nodeUri = ObjectExplorerUtils.getNodeUri(node);
+        await this._connectionManager.disconnect(nodeUri);
         if (!isDisconnect) {
             const index = this._rootTreeNodeArray.indexOf(node, 0);
             if (index > -1) {
                 this._rootTreeNodeArray.splice(index, 1);
             }
+
+        } else {
+            node.nodeType = Constants.disconnectedServerLabel;
+            node.contextValue = Constants.disconnectedServerLabel;
+            node.sessionId = undefined;
+            if (!(<IConnectionProfile>node.connectionCredentials).savePassword) {
+                node.connectionCredentials.password = '';
+            }
+            // make a new node to show disconnected behavior
+            let disconnectedNode = new TreeNodeInfo(node.label, Constants.disconnectedServerLabel,
+                node.collapsibleState, node.nodePath, node.nodeStatus, Constants.disconnectedServerLabel,
+                undefined, node.connectionCredentials, node.parentNode);
+            this.updateNode(disconnectedNode);
+            this._currentNode = disconnectedNode;
+            this._treeNodeToChildrenMap.set(this._currentNode, [new ConnectTreeNode(this._currentNode)]);
         }
-        const nodeUri = ObjectExplorerUtils.getNodeUri(node);
-        await this._connectionManager.disconnect(nodeUri);
         this._nodePathToNodeLabelMap.delete(node.nodePath);
         this.cleanNodeChildren(node);
-        if (isDisconnect) {
-            this._treeNodeToChildrenMap.set(this._currentNode, [new ConnectTreeNode(this._currentNode)]);
+    }
+
+    public async removeStaleConnectionNodes(staleConnections: IConnectionCredentials[]): Promise<void> {
```

Comment:
>removeStaleConnectionNodes [](start = 17, length = 26)

I'd suggest calling this something like removeConnectionNodes instead - there's nothing in the logic that makes this specifically for removing stale nodes only and something like this might be useful elsewhere